src/app/(protected)/prompts/[bookPromptId]/page.tsx

```diff
@@ -51,20 +47,6 @@ export default function PromptPage({
     }
   }, [bookPrompt, bookPromptId, isNew, loadBookPrompt, router]);
 
-  useEffect(() => {
-    if (bookPrompt) {
-      const recommendationsWithReviews = bookPrompt.bookRecommendations.map(
-        (recommendation) => {
-          const bookReview = bookReviews.find(
-            (review) => review.bookId === recommendation.book.id,
-          );
-          return { ...recommendation, bookReview };
-        },
-      );
-      setRecommendationsWithReviews(recommendationsWithReviews);
-    }
-  }, [bookPrompt, bookReviews]);
-
```

Comment:
We can move this logic below to build up the data on render, essentially doing the same thing but making the logic a little easier to understand, IMO